cmake/kokkos_tpls.cmake

```diff
@@ -37,6 +37,15 @@ ELSE()
 ENDIF()
 KOKKOS_TPL_OPTION(ROCM    ${ROCM_DEFAULT})
 KOKKOS_TPL_OPTION(ROCTHRUST ${ROCTHRUST_DEFAULT})
+if(Kokkos_ENABLE_ROCTHRUST)
+  include(CheckCXXSourceCompiles)
+  check_cxx_source_compiles("
+    #include <ios>
+    static_assert(_GLIBCXX_RELEASE < 9);
+    "
+    Kokkos_ENABLE_IMPL_SKIP_NO_RTTI_FLAG
+  )
```

Comment:
@ndellingwood can you double check we enter that IF branch and that the option is being set?

Reply:
```suggestion
  )
  message(FATAL_ERROR "Kokkos_ENABLE_IMPL_SKIP_NO_RTTI_FLAG=${Kokkos_ENABLE_IMPL_SKIP_NO_RTTI_FLAG}"
```

Reply:
From the CMakeCache: `Kokkos_ENABLE_IMPL_SKIP_NO_RTTI_FLAG:INTERNAL=` is unset

I tried a couple other things:
* The https://cmake.org/cmake/help/latest/module/CheckCXXSourceCompiles.html docs say a main function is required in the code to test, so I added that but no change from `check_cxx_source_compiles`
* I pulled out the code to test manually and it compiled an passed without issue:
```
// code snip: check_static_assert.cpp
#include <ios>
int main() {static_assert(_GLIBCXX_RELEASE < 9); return 0;}

// compiled as
hipcc --std=c++17 check_static_assert.cpp
```

Reply:
Added a print statement, `_GLIBCXX_RELEASE` is recognized as 8, I'll try changing the check to a runtime check on the `_GLIBCXX_RELEASE` (if it is defined)

Reply:
Oh maybe because it is not linking against rocthrust...

Reply:
@dalg24 a change like below worked to set the flag as expected:
```diff
diff --git a/cmake/kokkos_tpls.cmake b/cmake/kokkos_tpls.cmake
index b96bc7744..d88e001da 100644
--- a/cmake/kokkos_tpls.cmake
+++ b/cmake/kokkos_tpls.cmake
@@ -41,11 +41,18 @@ if(Kokkos_ENABLE_ROCTHRUST)
   include(CheckCXXSourceCompiles)
   check_cxx_source_compiles("
     #include <ios>
-    static_assert(_GLIBCXX_RELEASE < 9);
+    int main() {
+     #if defined(_GLIBCXX_RELEASE)
+     if (_GLIBCXX_RELEASE <= 8)
+       return -1;
+     #endif
+     return 0;
+    }
     "
     Kokkos_ENABLE_IMPL_SKIP_NO_RTTI_FLAG
   )
 endif()
+  MESSAGE(STATUS "**CHECK RTTI FLAG: Kokkos_ENABLE_IMPL_SKIP_NO_RTTI_FLAG=${Kokkos_ENABLE_IMPL_SKIP_NO_RTTI_FLAG}")
 
 IF(KOKKOS_ENABLE_SYCL)
   SET(ONEDPL_DEFAULT ON)
```

Output message: `-- **CHECK RTTI FLAG: Kokkos_ENABLE_IMPL_SKIP_NO_RTTI_FLAG=1`

Reply:
Yours would always compile would it not?

Reply:
oh that's it the main!!

Reply:
@dalg24 yes the change I tested will always compile, I shifted the check to a runtime check. The static_assert never triggered a compilation error for me, even in an isolated cpp test file

Reply:
Oop, the check was never supposed to fail the static_assert check for `_GLIBCXX_RELEASE == 8` 🤭

Reply:
it will definitely fail if you make hipcc pick up a more recent version of GCC libstdc++
It must fail at compile time
I updated the PR

Reply:
Yes, your newest commit resolves the issue 👍 
`Kokkos_ENABLE_IMPL_SKIP_NO_RTTI_FLAG=1` is set